src/modules/Elsa.ProtoActor/Services/ProtoActorWorkflowRuntime.cs

```diff
@@ -249,10 +249,51 @@ public async Task<WorkflowExecutionResult> ExecuteWorkflowAsync(WorkflowMatch ma
     }
 
     /// <inheritdoc />
-    public async Task CancelWorkflowAsync(string workflowInstanceId, CancellationToken cancellationToken)
+    public async Task<CancellationResult> CancelWorkflowAsync(string workflowInstanceId, CancellationToken cancellationToken)
     {
+        var filter = new WorkflowInstanceFilter
+        {
+            Id = workflowInstanceId
+        };
+
+        var instance = await _workflowInstanceStore.FindAsync(filter);
+        if (instance is null)
+            return new CancellationResult(false, FailureReason.NotFound);
+        
         var client = _cluster.GetNamedWorkflowGrain(workflowInstanceId);
-        await client.Cancel(cancellationToken);
+        var result = await client.Cancel(cancellationToken);
+        return new CancellationResult(result?.Result ?? false);
+    }
+
+    public async Task<int> CancelWorkflowAsync(IEnumerable<string> workflowInstanceIds, CancellationToken cancellationToken = default)
+    {
+        var tasks = workflowInstanceIds.Select(id => CancelWorkflowAsync(id, cancellationToken)).ToList();
+        await Task.WhenAll(tasks);
+        return tasks.Count(t => t.Result.Result);
+    }
+
+    public async Task<int> CancelWorkflowByDefinitionVersionAsync(string definitionVersionId, CancellationToken cancellationToken = default)
+    {
+        var filter = new WorkflowInstanceFilter
+        {
+            DefinitionVersionId = definitionVersionId,
+            WorkflowStatus = WorkflowStatus.Running
+        };
+        var instances = (await _workflowInstanceStore.FindManyAsync(filter, cancellationToken)).ToList();
+        await Task.WhenAll(instances.Select(instance => CancelWorkflowAsync(instance.Id, cancellationToken)));
+        
+        return instances.Count;
+    }
+
+    public async Task<int> CancelWorkflowByDefinitionAsync(string definitionId, VersionOptions versionOptions, CancellationToken cancellationToken = default)
+    {
+        // Shouldn't we get possible multiple definitions here?
+        var workflowDefinition = await _workflowDefinitionService.FindAsync(definitionId, versionOptions, cancellationToken);
+        if (workflowDefinition is null)
+            return 0;
+
+        var result = await CancelWorkflowByDefinitionVersionAsync(workflowDefinition.Id, cancellationToken);
+        return result;
     }
```

Comment:
Since these 3 methods seem to be mostly conveniences to client code, it seems to me that it would be better to extract these as extension methods on IWorkflowRuntime, so that the DefaultWorkflowRuntime automatically gets these methods too without having to implement them twice.